mixins/period_mixin.py

```diff
@@ -36,7 +36,7 @@ def save(self):
         if (self.end_date is not None) and (self.end_date < self.start_date):
             raise ValidationError('End date cannot be before start date')
 
-        return super(PeriodMixin, self).save()
+        return super(PeriodMixin, self).save(*args, **kwargs)
```

Comment:
Should have pointed this earlier, but in Python 3 you can write `super()` instead of this.
```suggestion
        return super().save(*args, **kwargs)
```